pkg/storage/mvcc.go

```diff
@@ -4850,7 +4856,7 @@ func MVCCGarbageCollect(
 					updateStatsForInline(ms, gcKey.Key, metaKeySize, metaValSize, 0, 0)
 					ms.AgeTo(timestamp.WallTime)
 				} else {
-					ms.Add(updateStatsOnGC(gcKey.Key, metaKeySize, metaValSize, meta, meta.Timestamp.WallTime))
+					ms.Add(updateStatsOnGC(gcKey.Key, metaKeySize, metaValSize, meta, realKeyChanged.WallTime))
```

Comment:
It's unclear to me why this isn't hitting the range tombstone branch above. Could you give an example and explanation of the case this covers?

Reply:
Isn't it more in spirit on what this return value represents? I think those values are equal in this branch, but i reverted that back.

Reply:
Huh, I thought that was what fixed the issue, because the tests passed with only this change. Maybe I got something wrong, I'll have another look.